src/jit/compiler.cpp

```diff
@@ -847,8 +861,13 @@ var_types Compiler::getArgTypeForStruct(CORINFO_CLASS_HANDLE clsHnd,
 
 #elif defined(_TARGET_ARM64_)
 
-                // Structs that are pointer sized or smaller should have been handled by getPrimitiveTypeForStruct
-                assert(structSize > TARGET_POINTER_SIZE);
+#if !defined(_TARGET_UNIX_)
+                if (!isVarArg)
```

Comment:
I don't see why this is needed. It should still be the case that `getPrimitiveTypeForStruct` should return a primitive type for anything <= `TARGET_POINTER_SIZE`, right?

Reply:
You are correct, thank you for catching this I do not know what I was thinking at the time.